classes/plugins/PKPPubIdPluginHelper.inc.php

```diff
@@ -195,11 +195,9 @@ function clearPubId($contextId, $pubIdPlugInClassName, $pubObject) {
 				if (get_class($pubIdPlugin) == $pubIdPlugInClassName) {
 					// clear the pubId:
 					// delete the pubId from the DB
-					$pubObjectType = $pubIdPlugin->getPubObjectType($pubObject);
-					$daos = $pubIdPlugin->getDAOs();
-					$dao = $daos[$pubObjectType];
+					$dao = $pubObject->getDAO();
 					$pubObjectId = $pubObject->getId();
-					if ($pubObjectType == 'SubmissionFile') {
+					if (is_a($pubObject, 'SubmissionFile')) {
```

Comment:
OK. 
I am not 100% sure, but somehow I think I had to do it this way, at least at that time, when I was implementing it some time ago. I will double check it all and change accordingly... THANKS!!!